src/test/ui/closures/issue-48109.rs

```diff
@@ -0,0 +1,14 @@
+// check-pass
+fn useful(i: usize) -> usize {
+    i
+}
+
+fn useful2(i: usize) -> usize {
+    i
+}
+
+fn main() {
+    for f in &[useful, useful2, |x| x] {
```

Comment:
What I meant above was that I would like to see tests like

```rust
type FnPointer = fn(usize) -> usize;
...
for f in [useful, useful2, |x| x] { .. } // already exists
for f in [useful as FnPointer, useful2, |x| x] { .. } // add these
for f in [useful, useful2 as FnPointer, |x| x] { .. }
for f in [useful, useful2, (|x| x) as FnPointer] { .. }
for f in [useful as FnPointer, useful2 as FnPointer, |x| x] { .. }
```

and then the same set of tests, but where we store the closure in a variable first:

```rust
let c = |x| x;
for f in [useful, useful2, c] { .. }
for f in [useful, useful2, c as FnPointer] { .. }
for f in [useful, useful2 as FnPointer, c] { .. }
for f in [useful as FnPointer, useful2, c] { .. }
for f in [useful as FnPointer, useful2 as FnPointer, c] { .. }
```

and perhaps a test like

```rust
let c = |x| x;
let c_pointer: FnPointer = c;
assert_eq!(c_pointer(22), 22);
```

I think if we're going to make improvements here, we should try to make uniform improvements.

Reply:
Contributing to rustc makes me more professional. XD